cmd/update/volume_move.go

```diff
@@ -108,14 +105,43 @@ func (c *volumeMoveCommand) runWithCtx(ctx context.Context, cmd *cobra.Command,
 	}
 
 	if !isMovingMaster && hasReplicaOnTargetNode {
-		c.display.TextMessage(ctx, c.writer, "volume already has a replica on both [source] and [destination] nodes, nothing to do")
+		c.display.TextMessage(ctx, c.writer, "Volume already has a replica on both [source] and [destination] nodes, nothing to do")
 		return nil
 	}
 
+	// after we know a valid scenario has been triggered we return what will
+	// happen back to the user in a user-friendly msg
+	if isMovingMaster {
+		if hasReplicaOnTargetNode {
+			c.display.TextMessage(ctx, c.writer, "Detected: promoting replica")
+		} else {
+			c.display.TextMessage(ctx, c.writer, "Detected: moving master")
+		}
+	} else {
+		c.display.TextMessage(ctx, c.writer, "Detected: moving replica")
+	}
+
 	originalReplicaCount := len(vol.Replicas)
 
 	// create a new replica on the [destination] node if there isn't one yet
 	if !hasReplicaOnTargetNode {
+		c.display.TextMessage(ctx, c.writer, `Creating a new replica on [destination] node.
+
+The duration of syncing operations depend on the rate of data transfers and network bandwidth.
+A default timeout of 30min is in place but it should be adjusted with the global --timeout flag to adapt to both volume and environment.
+`)
+
+		// when we need to sync a new replica, force the use of a timeout with
+		// a minimum of 30 minutes (if not already set) to avoid unecessary
+		// errors due to not using a long enough timeout.
+		//
+		// most users will be moving pretty large volumes of data anyways.
+		if deadline, _ := ctx.Deadline(); time.Until(deadline) < time.Minute*30 {
```

Comment:
How does this work if no deadline is set? 

Working it through in my head: `deadline` will be the zero time value, but then what does `time.Until(deadline)` return?

Reply:
There's a default one in place (15s I believe)